### PR TITLE
Docs Feedback widget

### PR DESCRIPTION
**Pull request by marcysutton**
## Description

Introducing a feedback widget to the docs section, with code integrated from [@jlengstorf's feedback widget repo](https://github.com/jlengstorf/gatsby-feedback-widget). We want to make it easy to collect feedback about docs from users in the moment, and by creating our own widget we will control the data (as opposed to a third-party service).

This PR adds two new dependencies for managing state:

- `xstate`
- `@xstate/react`

It would be super cool to add some tests for this, primarily to assert focus management in after view changes so we know they continue to work. Hopefully I'll get some time to work on that...but I didn't want to hold up reviewing of this in the meantime. 

## TODO:
Test with NVDA on IE11 and Firefox. The master branch is currently failing to build for me so I can't test it with Parallels.

### CODE REVIEW

**Comment by greglobinski** [comment]
I've changed `textarea` height to fix widget height

before

Screenshot from 2019-04-24 19-28-12

after

Screenshot from 2019-04-24 19-40-48

**Comment by pieh** [comment]
For people who want to try it - here's preview https://5cc089e52d11ebeccaf8db72--gatsby-org-pieh-previews.netlify.com/docs/ (just a note - it doesn't contain changes made in last 2 commits by Flo and Greg)

**Comment by fk** [comment]
Just pushed the missing design tokens stuff and fixed layout for screens below `breakpoints.lg` (horizontal whitespace, max-width matching content), and merged master to adjust things for #13388.

@greglobinski I had to do some gymnastics for smaller screens/max-width, and changed the `textarea` width from `99%` to `100%`, and removed `overflow-y: auto`. I think everything is still working fine (checked animations, etc.) — could you take another look though please? Don't want to break anything, and not sure if I am aware of everything that needs checking.

Here's how it looks now:

image

image

image

image

image

**Comment by greglobinski** [comment]
@fk Looking good, thank you

**Comment by prestonso** [comment]
I know we have the `?` affordance that doubles as a question mark currently, but there should be a question mark at the end of the phrase as well:

```
Was this doc helpful to you?
```

instead of:

```
Was this doc helpful to you
```

**Comment by greglobinski** [comment]
Actually, there is a hidden question mark for screen readers

```
 <ToggleButtonLabel>
              Was this doc helpful to you
              <ScreenReaderText>?</ScreenReaderText>
            </ToggleButtonLabel>
``` 

so we can show it if we want.

**Comment by marcysutton** [comment]
Adding a visible question mark in the button text makes sense to me. 

We should also factor this widget in to the docs internationalization effort–would we hide it for other languages? Or collect feedback and get it translated?

**Comment by pieh** [comment]
Few notes about visible extra question mark - I think it looks pretty weird in desktop variant (like double question mark):
Screenshot 2019-05-02 at 18 13 19

It does seem ok for narrow viewports:
Screenshot 2019-05-02 at 18 13 13
Screenshot 2019-05-02 at 18 13 05

I can implement it to show only for narrow viewports but it will feel pretty inconsistent to me. Thoughts?

**Comment by greglobinski** [comment]
@marcysutton  @pieh @prestonso 

What about such a solution? We always show question at the end of the question, but instead of Question mark icon we show the Chat icon?

Screenshot from 2019-05-06 16-27-53
Screenshot from 2019-05-06 16-26-26

**Comment by marcysutton** [comment]
Ooh, I like that! I think using a chat/message icon solves the problem very well.

**Comment by fk** [comment]
☝️☝️☝️

**Comment by muescha** [comment]
is this a one time glitch or is it fixed?
97C96E2E-7694-47E7-B4F5-251C9471DE3B

**Comment by greglobinski** [comment]
@muescha 
> Found in screenshot above: is this a one time glitch or is it fixed?

What exactly are you asking about?

**Comment by muescha** [comment]
The "poor" icon in this screenshots is shiftet down and not visible

**Comment by greglobinski** [comment]
> The "poor" icon in this screenshots is shiftet down and not visible

Ok, that's the intentional `hover` state of the input.

File: www/src/components/feedback-widget/buttons.js [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+  0% {
+    transform: translateX(${space[1]}) rotate(0deg);
+  }
+  100% {
+    transform: translateX(${space[1]}) rotate(360deg);
+  }
+`
+
+export const focusStyle = css`
+  box-shadow: 0 0 0 0.12rem ${colors.accent};
```
Comment (marcysutton) [inline www/src/components/feedback-widget/buttons.js]: This focus style still isn't working in Safari. I played around with it a bit and opted to change it to use `outline` and `outline-offset` instead:
```
outline: 2px solid ${colors.accent};
outline-offset: -2px;
```
https://codepen.io/marcysutton/pen/WBxEom

**Comment by marcysutton** [comment]
This is super close! The only issue I ran into is if you hit the enter key on the smiley radio buttons, I don't think it goes through the submit flow....it seems to cancel and reset back to the widget toggle button instead.

**Comment by muescha** [comment]
Where is the server side code repo for the mutation `submitFeedback`

**Comment by jlengstorf** [comment]
@muescha the API source is here. It won't run without setting up a Prisma instance, though — take a look at the `.env.EXAMPLE` file for details. https://github.com/gatsbyjs/api.gatsbyjs.org/blob/master/src/graphql/public-resolvers.js

File: www/src/components/feedback-widget/question-mark-icon.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from "react"
+
+export default () => (
+  <svg
```
[the remark this answers is not in the document]
Comment (marcysutton) [reply www/src/components/feedback-widget/question-mark-icon.js]: Good catch. I've removed the extra code!

**Comment by marcysutton** [comment]
This is ready to go, assuming the form endpoint works. @jlengstorf if you think we need to test that more fully I could use some detailed instructions. Otherwise I think this is ready to merge!

**Comment by marcysutton** [comment]
It works now that I have the API endpoint set up locally (thanks @jlengstorf!). Excited to start collecting data!

**Comment by fk** [comment]
Yay! 🎉

**Pull request merged**